bundles/org.eclipse.core.jobs/src/org/eclipse/core/runtime/jobs/ISchedulableOperation.java

```diff
@@ -0,0 +1,31 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Joerg Kubitz.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Joerg Kubitz - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.core.runtime.jobs;
+
+/**
+ * An interface to mark an Operation that needs an ISchedulingRule.
+ *
+ * @since 3.13
+ *
+ */
+public interface ISchedulableOperation {
+	/**
+	 * @return an ISchedulingRule that the operation will acquire - if any. A caller
+	 *         can make sure that this ISchedulableOperation is locked before
+	 *         calling the operation. Returns null if no rule needed.
+	 *
+	 * @since 3.13
+	 */
```

Comment:
For Javadoc on method/class, please:
- use "@link" for all public type names mentioned
- use "{@code null}"
- `this ISchedulableOperation is locked` is not correct, should be `this ISchedulingRule is locked in current thread`
- regarding `A caller can make sure` - this is too vague. So **should** the operation begin the rule, should it not, can or can not? Better to say something like "if the rule is non null, the caller will make sure that IJobManager#beginRule(ISchedulingRule, IProgressMonitor) will be called before the operation and endRule() after.

Reply:
ok, done